forc-plugins/forc-client/src/op/call/call_function.rs

```diff
@@ -0,0 +1,789 @@
+use fuel_abi_types::abi::{program::ProgramABI, unified_program::UnifiedProgramABI};
+use fuels::{
+    accounts::ViewOnlyAccount,
+    programs::calls::{
+        receipt_parser::ReceiptParser,
+        traits::{ContractDependencyConfigurator, TransactionTuner},
+        ContractCall,
+    },
+};
+use fuels_core::{
+    codec::{
+        encode_fn_selector, log_formatters_lookup, ABIDecoder, ABIEncoder, DecoderConfig,
+        EncoderConfig, LogDecoder,
+    },
+    types::{
+        bech32::Bech32ContractId,
+        param_types::ParamType,
+        transaction::Transaction,
+        transaction_builders::{BuildableTransaction, ScriptBuildStrategy, VariableOutputPolicy},
+        ContractId,
+    },
+};
+use std::{collections::HashMap, path::PathBuf};
+use url::Url;
+
+/// Calls a contract function with the given parameters
+pub async fn call_function(
+    contract_id: ContractId,
+    abi: Either<PathBuf, Url>,
+    function: FuncType,
+    function_args: Vec<String>,
+    cmd: cmd::Call,
+) -> Result<CallResponse> {
+    let cmd::Call {
+        node,
+        mode,
+        caller,
+        call_parameters,
+        gas,
+        show_receipts,
+        output,
+        external_contracts,
+        ..
+    } = cmd;
+
+    // Load ABI (already provided in the operation)
+    let abi_str = super::load_abi(&abi).await?;
+    let parsed_abi: ProgramABI = serde_json::from_str(&abi_str)?;
+    let unified_program_abi = UnifiedProgramABI::from_counterpart(&parsed_abi)?;
+
+    let cmd::call::FuncType::Selector(selector) = function;
+
+    // Setup connection to node
+    let (wallet, tx_policies, base_asset_id) = super::setup_connection(&node, caller, &gas).await?;
+    let call_parameters = cmd::call::CallParametersOpts {
+        asset_id: call_parameters.asset_id.or(Some(base_asset_id)),
+        ..call_parameters
+    };
+
+    // Get function definition from ABI
+    let type_lookup = unified_program_abi
+        .types
+        .iter()
+        .map(|decl| (decl.type_id, decl.clone()))
+        .collect::<HashMap<_, _>>();
+
+    // Find the function in the ABI
+    let abi_func = unified_program_abi
+        .functions
+        .iter()
+        .find(|abi_func| abi_func.name == selector)
+        .ok_or_else(|| anyhow!("Function '{}' not found in ABI", selector))?;
+
+    // Validate number of arguments
+    if abi_func.inputs.len() != function_args.len() {
+        bail!(
+            "Number of arguments does not match number of parameters in function signature; expected {}, got {}", 
+            abi_func.inputs.len(),
+            function_args.len()
+        );
+    }
+
+    // Parse function arguments to tokens
+    let tokens = abi_func
+        .inputs
+        .iter()
+        .zip(function_args)
+        .map(|(type_application, arg)| {
+            let param_type = ParamType::try_from_type_application(type_application, &type_lookup)
+                .expect("Failed to convert input type application");
```

Comment:
We are propagating the error for finding the `abi_func` but panicking for this one, is there a reason for that. I think it might be nicer if we are propagating all the errors.

Reply:
https://github.com/FuelLabs/sway/pull/7056/commits/ee3bd29e5fccc6d3eb9115bc3e5babf5f430da96